diff --git a/sdk/userspace/include/fpga_mgmt.h b/sdk/userspace/include/fpga_mgmt.h
index 57ad437..7789bda 100644
--- a/sdk/userspace/include/fpga_mgmt.h
+++ b/sdk/userspace/include/fpga_mgmt.h
@@ -26,6 +26,9 @@
  *
  * @returns 0 on success, non-zero on error
  */
+#ifdef __GPLUSPLUS__
+extern "C"
+#endif
 int fpga_mgmt_init(void);
 
 /**
@@ -84,6 +87,9 @@ struct fpga_mgmt_image_info {
  * @param[in]  flags    set flags for for metrics retrieval options
  * @returns 0 on success, non-zero on error
  */
+#ifdef __GPLUSPLUS__
+extern "C"
+#endif
 int fpga_mgmt_describe_local_image(int slot_id,
 	struct fpga_mgmt_image_info *info, uint32_t flags);
 
diff --git a/sdk/userspace/include/fpga_pci.h b/sdk/userspace/include/fpga_pci.h
index b36b326..31a5fdb 100644
--- a/sdk/userspace/include/fpga_pci.h
+++ b/sdk/userspace/include/fpga_pci.h
@@ -59,6 +59,9 @@ typedef int pci_bar_handle_t;
  * Initialize the pci library.
  * @returns 0 on success, non-zero on error
  */
+#ifdef __GPLUSPLUS__
+extern "C"
+#endif
 int fpga_pci_init(void);
 
 /**
@@ -73,6 +76,9 @@ int fpga_pci_init(void);
  *
  * @returns 0 on success, non-zero on error
  */
+#ifdef __GPLUSPLUS__
+extern "C"
+#endif
 int fpga_pci_attach(int slot_id, int pf_id, int bar_id, uint32_t flags,
     pci_bar_handle_t *handle);
 
@@ -91,6 +97,9 @@ enum {
  *                     the memory space to detach
  * @returns 0 on success, non-zero on error
  */
+#ifdef __GPLUSPLUS__
+extern "C"
+#endif
 int fpga_pci_detach(pci_bar_handle_t handle);
 
 /**
@@ -101,6 +110,9 @@ int fpga_pci_detach(pci_bar_handle_t handle);
  * @param[in]  value   value to write to the register
  * @returns 0 on success, non-zero on error
  */
+#ifdef __GPLUSPLUS__
+extern "C"
+#endif
 int fpga_pci_poke(pci_bar_handle_t handle, uint64_t offset, uint32_t value);
 
 /**
@@ -134,6 +146,9 @@ int fpga_pci_write_burst(pci_bar_handle_t handle, uint64_t offset,
  * @param[out] value   value read from the register (32-bit)
  * @returns 0 on success, non-zero on error
  */
+#ifdef __GPLUSPLUS__
+extern "C"
+#endif
 int fpga_pci_peek(pci_bar_handle_t handle, uint64_t offset, uint32_t *value);
 
 /**
diff --git a/sdk/userspace/include/utils/log.h b/sdk/userspace/include/utils/log.h
index 60cec19..dfd1bd1 100644
--- a/sdk/userspace/include/utils/log.h
+++ b/sdk/userspace/include/utils/log.h
@@ -105,6 +105,9 @@ __printf(1, 2) int log_init_prefix(const char *format, ...);
 int log_attach(const struct logger *logger, void *opaque, unsigned int flags);
 
 /** Log an error message (do not call directly!). */
+#ifdef __GPLUSPLUS__
+extern "C"
+#endif
 __printf(4, 5) void _log_error(const char *file, int line, const char *func,
 			       const char *format, ...);
 
